src/App.tsx

```diff
@@ -8,53 +9,102 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getUserPosts } from './api/posts';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [postsIsLoading, setPostsIsLoading] = useState(false);
+  const [hasPostsError, setHasPostsError] = useState('');
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  const [chosenPost, setChosenPost] = useState<Post | null>(null);
 
-              <Loader />
+  const hasNoPosts =
+    selectedUser && !postsIsLoading && !hasPostsError && !posts.length;
```

Comment:
The `hasNoPosts` variable is not correctly initialized. It should be a boolean expression, but currently, it is missing the assignment operator `=`. It should be `const hasNoPosts = selectedUser && !postsIsLoading && !hasPostsError && !posts.length;`.